controllers/process.go

```diff
@@ -146,7 +146,35 @@ func (r *ServicefenceReconciler) ReconcileService(req ctrl.Request) (ctrl.Result
 	r.reconcileLock.Lock()
 	defer r.reconcileLock.Unlock()
 
-	return r.refreshFenceStatusOfService(ctx, nil, req.NamespacedName)
+	r.svcSelectorCache.Lock()
+	defer r.svcSelectorCache.Unlock()
+
+	nsName := types.NamespacedName{Namespace: req.Namespace}
+	svc := &corev1.Service{}
+	err := r.Client.Get(ctx, req.NamespacedName, svc)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// delete svc event
+			nsName.Name = r.svcSelectorCache.Data[req.NamespacedName]
+			delete(r.svcSelectorCache.Data, req.NamespacedName)
+			return r.refreshFenceStatusOfService(ctx, nil, nsName)
+		} else {
+			log.Errorf("get service %s error, %+v", req.NamespacedName, err)
+			return reconcile.Result{}, err
+		}
+	} else {
+		// add or update svc event
+		if sel, ok := r.svcSelectorCache.Data[req.NamespacedName]; ok && sel != svc.Spec.Selector[r.env.Config.Global.Service] {
```

Comment:
The service selector's labels need to be matched at the same time, so here we need to use the whole selector as a filtering condition, i.e. workloadselector